packages/qwik/src/prefetch-service-worker/setup.ts

```diff
@@ -46,3 +48,7 @@ export const setupServiceWorker = (swScope: ServiceWorkerGlobalScope) => {
     event.waitUntil(swScope.clients.claim());
   });
 };
+
+function isQwikMessages(msg: any): msg is SWMessages {
+  return msg[0] && Object.values(MsgType).includes(msg[0][0]);
```

Comment:
every message that comes in you generate an array of all the values in the enum. How about renaming the enum values so they are lowercase and you can do `!!MsgType[msg[0][0]]`?

Reply:
Ah yes you’re right. Hmm, the tricky part is that some of the enum values can’t be valid enum keys, like `graph-url`. We could generate the array once outside of the function. I’m not sure what a better solution could be, what do you think?